auslib/test/admin/views/test_releases.py

```diff
@@ -783,3 +783,22 @@ def testSettings(self):
         ret = self._get('/releases/settings')
         self.assertStatusCode(ret, 200)
         self.assertEqual(json.loads(ret.data), json.loads(data['blob']))
+
+
+class TestReadOnlyView(ViewTest, JSONTestMixin):
+
+    def testReadOnlyGet(self):
+        ret = self._get('/releases/b/read_only')
+        is_read_only = dbo.releases.t.select(dbo.releases.name == 'b').execute().first()['read_only']
+        self.assertEqual(json.loads(ret.data)['read_only'], is_read_only)
+
+    def testReadOnlySetTrue(self):
+        data = dict(read_only=True, product='Firefox', data_version=1)
+        self._put('/releases/b/read_only', data=data)
+        read_only = dbo.releases.isReadOnly(name='b')
+        self.assertEqual(read_only, True)
+
+    def testReadOnlySetFalseWithoutPermissionForProduct(self):
+        data = dict(read_only=False, product='Firefox', data_version=1)
+        ret = self._put('/releases/b/read_only', username='bob', data=data)
+        self.assertStatusCode(ret, 401)
```

Comment:
These tests look fine, but we need a few more. At least:
* Verify that an admin can unset the read only flag
* Verify that a non-admin user can *set* the read only flag (testReadOnlySetTrue uses an admin user). This probably required tweaking the permissions in https://github.com/mozilla/balrog/blob/master/auslib/test/admin/views/base.py#L15.
* Verify that a non-admin user who can set the read-only flag can't unset it. I think this is broken at the moment because of the missing "Return" statement in the view code. This case should probably expect a 403.
* Verify that other views fail when trying to modify a read-only release. These should probably be in existing classes.